src/BabyLion_Management/BabyLion_SearchService.java

```diff
@@ -0,0 +1,20 @@
+package BabyLion_Management;
+import java.util.*;
+
+//검색관련 클래스
+public class BabyLion_SearchService {
+    private Map<String, BabyLion_Info> lions;
+    private Scanner scanner;
+
+    public BabyLion_SearchService(Map<String, BabyLion_Info> lions, Scanner scanner) {
+        this.lions = lions;
+        this.scanner = scanner;
+    }
+
+    //학번으로 아기사자 검색, 해당 학번에 맞는 아기사자 객체를 반환
+    public BabyLion_Info searchLion() {
+        System.out.print("학번을 입력하세요 : ");
+        String searchID = scanner.nextLine();
+        return lions.get(searchID);
+    }
```

Comment:
`BabyLion_SearchService` 내에서 사용자의 입력을 받기때문에 Scanner 객체를 필드로 선언해줘야 하는 소요가 생겼는데 해당 구조를 어떤 식으로 바꾸면 Scanner 객체를 굳이 멤버 변수로 선언하지 않고도 똑같이 동작할 수 있을까요?

Reply:
질문 감사드립니다!

해당 부분은 따로 입력을 담당하는 클래스를 하나 선언하면 좋을 것 같다고 생각했습니다.
해당 클래스에서 메소드를 static으로 구현하면, new로 선언할 필요 없이 바로 메소드를 호출할 수 있고, 입력받을 타입에 따라 따로 구현해서 사용하면 관리하기도 편할 것 같습니다. 또한 예외처리부분도 해당 클래스에서 바로 try-catch를 사용하면 될 것 같아서 더 좋을 것 같습니다. 그리고 main에서 종료될때 해당 scanner를 close해주면 더 좋을 것 같습니다.

Reply:
이 외에도 `searchLion()`을 호출하는 클래스에서 입력값을 받아 입력값만 인자로 전달한다면 불필요한 Scanner 객체 생성을 막을 수 있을 것 같습니다!

```suggestion
public BabyLion_Info searchLion(String searchId) {
        return lions.get(searchId);
}
```

Reply:
아.. 제일 쉬운 방법이 있었네요! 앞으로 스캐너도 많이 생각해봐야겠네요! 감사합니다.